pkg/dataobj/sections/logs/encoder.go

```diff
@@ -137,11 +139,17 @@ func (enc *encoder) Flush(w dataobj.SectionWriter) (int64, error) {
 // columns.
 func (enc *encoder) Reset() {
 	bufpool.PutUnsized(enc.data)
+	enc.tenant = ""
 	enc.data = nil
 	enc.curColumn = nil
 	enc.sortInfo = nil
 }
 
+// SetTenant sets the tenant who owns the streams section.
+func (enc *encoder) SetTenant(tenant string) {
```

Comment:
It doesn't look like this method is ever used for either encoder. Is that intentional for this PR or is it missing a call?

Reply:
It's intentional, I'm adding the changes from the feature branch incrementally. It gets used by the builder.